crates/red_knot/build.rs

```diff
@@ -28,36 +33,38 @@ fn zip_dir(directory_path: &str, writer: File) -> ZipResult<File> {
 
     for entry in walkdir::WalkDir::new(directory_path) {
         let dir_entry = entry.unwrap();
-        let relative_path = dir_entry.path();
-        let name = relative_path
+        let absolute_path = dir_entry.path();
+        let relative_path = absolute_path
             .strip_prefix(Path::new(directory_path))
-            .unwrap()
+            .unwrap();
+        if !predicate(relative_path) {
```

Comment:
What's the motivation for passing a relative path when the predicate logic makes the path absolute again by joining (and the extension part doesn't care about relative or absolute)